cmd/ami-cleaner/main.go

```diff
@@ -0,0 +1,92 @@
+package main
+
+import (
+	"github.com/trussworks/truss-aws-tools/internal/aws/session"
+	"github.com/trussworks/truss-aws-tools/pkg/amiclean"
+
+	"github.com/aws/aws-lambda-go/lambda"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	flag "github.com/jessevdk/go-flags"
+	"go.uber.org/zap"
+
+	"log"
+	"time"
+)
+
+// The Options struct describes the command line options available.
+type Options struct {
+	Delete        bool   `short:"D" long:"delete" description:"Actually purge AMIs (runs in dryrun mode by default)."`
+	RetentionDays int    `long:"days" default:"30" description:"Age of AMI in days before it is a candidate for removal."`
+	Branch        string `short:"b" long:"branch" required:"true" description:"Branch to purge. If the the --invert option is used, this is the branch NOT to operate on."`
+	Invert        bool   `short:"i" long:"invert" description:"Operate in inverted mode -- only purge AMIs that are NOT in the branch provided."`
+	Profile       string `short:"p" long:"profile" env:"PROFILE" required:"false" description:"The AWS profile to use."`
+	Region        string `short:"r" long:"region" env:"REGION" required:"false" description:"The AWS region to use."`
+	Lambda        bool   `long:"lambda" required:"false" env:"LAMBDA" description:"Run as an AWS Lambda function."`
+}
+
+var options Options
```

Comment:
So you're saying that instead of doing it this way, you'd change it so that there's no `var options Options` or `var logger *zap.Logger`, and change the cleanImages definition to something like:

```go
func cleanImages(o Options, l Logger) {
```
...and then calling it later with...
```go
cleanImages(options, *zap.Logger)
```
Looking at this code (which I admit I mostly just cut and pasted from @dynamike's rdsclean stuff), I can see the potential benefit of that but I'm not sure if Lambda works that way. I'll ask Mike about it when he's not at 30,000 feet I guess. :)

Reply:
Yeah, that is how I would do it. It's basically the same pattern you have going with the AMIClean struct, the logger et al are explicitly passed in when you create one rather than pulling from a global.